### PR TITLE
add basic gotests support

### PR DESCRIPTION
**Pull request by marcomayer**
This adds basic support for https://github.com/cweill/gotests. Hovering over a function doing :GoAddTest adds test for function in _test.go file. I used treesitter to get the function name.

Sorry for the bad lua code, I've very basic knowledge of it, also credits go to https://github.com/ray-x/go.nvim for the idea and part of the code.

### CODE REVIEW

File: lua/go/gotests.lua [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        table.insert(cmd, opt.gotests_template)
+        if string.len(opt.gotests_template_dir) > 0 then
+            table.insert(cmd, '-template_dir')
+            table.insert(cmd, opt.gotests_template_dir)
+        end
+    end
+
+    local funame = function_surrounding_cursor()
+    if funame == nil or funame == '' then
+        print('no function found')
```
Comment (crispgm) [inline lua/go/gotests.lua]: `output.show_error` here

File: lua/go/config.lua [hunk 2]
```diff
@@ -32,6 +32,8 @@ M.options = {
     tags_flags = { '-skip-unexported' },
     -- quick type
     quick_type_flags = { '--just-types' },
+    gotests_template = '', -- sets gotests -template parameter (see gotests for details)
```
Comment (crispgm) [inline lua/go/config.lua:36]: a separating comment should be here :)

**Comment by crispgm** [comment]
Looks quite good, please check the minor issues I submitted during review.

**Comment by marcomayer** [comment]
> 

Happy to! Just can't see your comments, maybe you didn't finish the review or there's some permission issue?

**Review by crispgm — CHANGES_REQUESTED** [review]

**Comment by crispgm** [comment]
> > 
> 
> Happy to! Just can't see your comments, maybe you didn't finish the review or there's some permission issue?

Yes, you are correct. I just submitted the review :)

**Comment by marcomayer** [comment]
> Yes, you are correct. I just submitted the review :)
Great, thanks! Just updated what was requested, thanks for pointing it out.

File: lua/go/config.lua [hunk 3]
```diff
@@ -48,6 +51,7 @@ M.tools = {
     },
     { name = 'gomodifytags', src = 'github.com/fatih/gomodifytags' },
     { name = 'quicktype', src = 'quicktype', pkg_mgr = 'npm' },
+    { name = 'gotests', src = 'github.com/cweill/gotests' },
```
Comment (crispgm) [inline lua/go/config.lua]: It looks like that the correct URL is: `github.com/cweill/gotests/gotests`

**Review by crispgm — CHANGES_REQUESTED** [review]

**Comment by crispgm** [comment]
@marcomayer And I have one more question: does `GoAddTest` need any parameters?

**Comment by crispgm** [comment]
@marcomayer just added a commit by myself to fix some stuffs

**Comment by marcomayer** [comment]
>

**Comment by marcomayer** [comment]
> @marcomayer And I have one more question: does `GoAddTest` need any parameters?

Nope, you just need to be in or hover at a function when calling it.

**Review by crispgm — APPROVED** [review]

**Comment by crispgm** [comment]
> > @marcomayer And I have one more question: does `GoAddTest` need any parameters?
> 
> Nope, you just need to be in or hover at a function when calling it.

Okay, I updated the `command` and then merged. Thanks!

**Pull request merged**